asf_core_data/pipeline/mcs/process/process_historical_mcs_installers.py

```diff
@@ -609,6 +608,11 @@ def update_effective_to_date(
         right_on=installations_match_vars,
     )
 
+    installer_data["effective_to"] = pd.to_datetime(installer_data["effective_to"])
+    installer_data["commissioning_date"] = pd.to_datetime(
+        installer_data["commissioning_date"]
+    )
+
```

Comment:
a welome addition. I wonder if it would've been better to convert in earlier processing steps of the respective datasets (i.e. on reading in the datasets)?

Reply:
I did previously encounter issues writing the parquet file as a column of mixed types cannot be written to parquet. So I did add a conversion bit after calling this function in [asf_daps process_mcs_flow.py](https://github.com/nestauk/asf_daps/blob/05dcc4aebb0e1e2ce202605302d9dd65bf330e67/asf_daps/flows/processed/mcs/process_mcs_flow.py#L101-L105)

Reply:
@sqr00t does it make mixed types to convert to datetime if there are NA values?

Reply:
It should now convert to pd.NA

Reply:
sorry I meant, does pd.Na and datetime values in the same column mean it would have mixed types and thus would not be able to write to parquet?

Reply:
There should be no issue

On write, I pass in a PyArrow schema object that checks and can optionally force conversion. pd.NA and datetime in the same column is convertible.